src/component/feature/variant/add-variant.jsx

```diff
@@ -167,10 +186,33 @@ function AddVariant({ showDialog, closeDialog, save, validateName, editVariant,
                     value={data.name}
                     error={error.name}
                     type="name"
-                    onChange={setName}
+                    onChange={setVariantValue}
                 />
                 <br />
                 <br />
+                <div
+                    style={{
+                        display: 'flex',
+                        alignItems: 'center',
+                    }}
+                >
+                    <Textfield
+                        floatingLabel
+                        label="Weight"
+                        name="weight"
+                        placeholder=""
+                        value={data.weight}
+                        error={error.weight}
+                        type="number"
+                        disabled={data.weightType !== 'fix'}
+                        onChange={setVariantValue}
+                    />
+                    <div>
+                        <Switch name="weightType" checked={isFixWeight} onChange={setVariantWeightType}>
+                            Fix
+                        </Switch>
```

Comment:
Maybe also clean up the view in general a bit, something along the lines of this:
![image](https://user-images.githubusercontent.com/158948/88976461-5af65880-d2bc-11ea-8225-ed78f0e2cbb7.png)